tests/RestfulCreateEntityTestCase.test

```diff
@@ -178,31 +178,73 @@ class RestfulCreateEntityTestCase extends RestfulCurlBaseTestCase {
 
   }
 
+  public function testFileUploadOptions() {
+    variable_set('restful_file_upload', TRUE);
+    variable_set('restful_file_upload_allow_anonymous_user', TRUE);
+
+    // Upload a permanent file.
+    $return = $this->uploadFile();
+    $this->assertEqual($return['code'], 200, 'File uploaded.');
+
+    $return_array = drupal_json_decode($return['body']);
+    $file = file_load($return_array['data'][0]['id']);
+
+    // Make sure it's the same file that was uploaded.
+    $this->compareFile($file);
+    // File should be permanent.
+    $this->assertEqual($file->status, FILE_STATUS_PERMANENT, 'File status is permanent.');
+    $usage = file_usage_list($file);
+    $this->assertEqual(reset($usage['restful']['files']), 1, 'File usage is registered by RESTful.');
+
+
+    // Upload a temporary file
+    //drupal_static_reset('restful_get_restful_handler');
+    cache_clear_all();
+print_r(array_keys(restful_get_restful_plugins()));
+    $handler = restful_get_restful_handler('files', 1, 0);
```

Comment:
You might need to set the [variable](https://github.com/RESTful-Drupal/restful/blob/7.x-1.x/plugins/restful/file/files_upload/1.0/files_upload__1_0.inc#L3) and clear menu cache